include/zephyr/drivers/fuel_gauge.h

```diff
@@ -28,90 +28,94 @@ extern "C" {
-#define FUEL_GAUGE_REMAINING_CAPACITY		FUEL_GAUGE_PRESENT_STATE + 1
-/** Remaining battery life time in minutes */
-#define FUEL_GAUGE_RUNTIME_TO_EMPTY		FUEL_GAUGE_REMAINING_CAPACITY + 1
-/** Remaining time in minutes until battery reaches full charge */
-#define FUEL_GAUGE_RUNTIME_TO_FULL		FUEL_GAUGE_RUNTIME_TO_EMPTY + 1
-/** Retrieve word from SBS1.1 ManufactuerAccess */
-#define FUEL_GAUGE_SBS_MFR_ACCESS		FUEL_GAUGE_RUNTIME_TO_FULL + 1
-/** Absolute state of charge (percent, 0-100) - expressed as % of design capacity */
-#define FUEL_GAUGE_ABSOLUTE_STATE_OF_CHARGE	    FUEL_GAUGE_SBS_MFR_ACCESS + 1
-/** Relative state of charge (percent, 0-100) - expressed as % of full charge capacity */
-#define FUEL_GAUGE_RELATIVE_STATE_OF_CHARGE		FUEL_GAUGE_ABSOLUTE_STATE_OF_CHARGE + 1
-/** Temperature in 0.1 K */
-#define FUEL_GAUGE_TEMPERATURE		    FUEL_GAUGE_RELATIVE_STATE_OF_CHARGE + 1
-/** Battery voltage (uV) */
-#define FUEL_GAUGE_VOLTAGE			FUEL_GAUGE_TEMPERATURE + 1
-/** Battery Mode (flags) */
-#define FUEL_GAUGE_SBS_MODE			FUEL_GAUGE_VOLTAGE + 1
-/** Battery desired Max Charging Current (mA) */
-#define FUEL_GAUGE_CHARGE_CURRENT		FUEL_GAUGE_SBS_MODE + 1
-/** Battery desired Max Charging Voltage (mV) */
-#define FUEL_GAUGE_CHARGE_VOLTAGE		FUEL_GAUGE_CHARGE_CURRENT + 1
-/** Alarm, Status and Error codes (flags) */
-#define FUEL_GAUGE_STATUS			FUEL_GAUGE_CHARGE_VOLTAGE + 1
-/** Design Capacity (mAh or 10mWh) */
-#define FUEL_GAUGE_DESIGN_CAPACITY		FUEL_GAUGE_STATUS + 1
-/** Design Voltage (mV) */
-#define FUEL_GAUGE_DESIGN_VOLTAGE		FUEL_GAUGE_DESIGN_CAPACITY + 1
-/** AtRate (mA or 10 mW) */
-#define FUEL_GAUGE_SBS_ATRATE			FUEL_GAUGE_DESIGN_VOLTAGE + 1
-/** AtRateTimeToFull (minutes) */
-#define FUEL_GAUGE_SBS_ATRATE_TIME_TO_FULL	FUEL_GAUGE_SBS_ATRATE + 1
-/** AtRateTimeToEmpty (minutes) */
-#define FUEL_GAUGE_SBS_ATRATE_TIME_TO_EMPTY	FUEL_GAUGE_SBS_ATRATE_TIME_TO_FULL + 1
-/** AtRateOK (boolean) */
-#define FUEL_GAUGE_SBS_ATRATE_OK		FUEL_GAUGE_SBS_ATRATE_TIME_TO_EMPTY + 1
-/** Remaining Capacity Alarm (mAh or 10mWh) */
-#define FUEL_GAUGE_SBS_REMAINING_CAPACITY_ALARM FUEL_GAUGE_SBS_ATRATE_OK + 1
-/** Remaining Time Alarm (minutes) */
-#define FUEL_GAUGE_SBS_REMAINING_TIME_ALARM	FUEL_GAUGE_SBS_REMAINING_CAPACITY_ALARM + 1
-/** Manufacturer of pack (1 byte length + 20 bytes data) */
-#define FUEL_GAUGE_MANUFACTURER_NAME		FUEL_GAUGE_SBS_REMAINING_TIME_ALARM + 1
-/** Name of pack (1 byte length + 20 bytes data) */
-#define FUEL_GAUGE_DEVICE_NAME			FUEL_GAUGE_MANUFACTURER_NAME + 1
-/** Chemistry (1 byte length + 4 bytes data) */
-#define FUEL_GAUGE_DEVICE_CHEMISTRY		FUEL_GAUGE_DEVICE_NAME + 1
-
-/** Reserved to demark end of common fuel gauge properties */
-#define FUEL_GAUGE_COMMON_COUNT FUEL_GAUGE_DEVICE_CHEMISTRY + 1
-/**
- * Reserved to demark downstream custom properties - use this value as the actual value may change
- * over future versions of this API
- */
-#define FUEL_GAUGE_CUSTOM_BEGIN FUEL_GAUGE_COMMON_COUNT + 1
+enum fuel_gauge_property {
```

Comment:
I had envisioned this as 3 different enums/types:
typedef enum fuel_gauge_common_property_t { 
     FG_COMMON_MIN = 0,
     (fg common properties),
    FG_COMMON_MAX };
typedef enum fuel_gauge_buffer_property_t {
   FG_BUFFER_MIN = FG_COMMON_MAX + 1,
  (buffer properties),
   FG_BUFFER_MAX }
(and the same for the custom properties)

This way, we get two things.  We can use the enum types in the functions and calls and get stronger type checking.  Also, we can use the MIN and MAX symbols for each group to do some checking in the driver to make sure we're in-bounds.
Downside is that it's likely a uint32_t size on most architectures.

I'm OK with it the way it is, just throwing it out there.

Reply:
Will do that in a follow-up PR. But wanted to make the switch to enums soon.